### PR TITLE
Yuhsuan/918 save restfreq

### LINKED ISSUE #918: apply a new RESTFRQ when exporting a subimage
the relevant frontend issue is https://github.com/CARTAvis/carta-frontend/issues/1653

The current subimage export adopts the same RESTFRQ defined the in header of the input cube. This feature is to support the ability to set a new rest frequency of the subimage. This backend part needs to apply the custom rest frequency to the header of the exported subimage.

### PR DESCRIPTION
**Pull request by YuHsuan-Hwang**
Closes #918:
Set custom rest freq when saving files. The image is saved as `casacore::PagedImage` when changing rest freq is required.

Companion PRs:
https://github.com/CARTAvis/carta-frontend/pull/1830
https://github.com/CARTAvis/carta-protobuf/pull/69

### CODE REVIEW

**Review by kswang1029 — APPROVED** [review]
tested 👍 @confluence @pford @markccchiang  please proceed code review.

**Review by markccchiang — APPROVED** [review]

File: src/Frame/Frame.cc [hunk 1]
```diff
@@ -1832,6 +1837,18 @@ void Frame::SaveFile(const std::string& root_folder, const CARTA::SaveFile& save
         return;
     }
 
+    if (change_rest_freq) {
+        casacore::CoordinateSystem coord_sys = image->coordinates();
+        casacore::String errorMsg("");
```
Comment (markccchiang) [inline src/Frame/Frame.cc]: Just a minor issue about the code style. `errorMsg` would be better as `error_msg`.

**Review by pford — CHANGES_REQUESTED** [review]
Please also update protobuf changelog

File: src/Frame/Frame.cc (hunk 1 above)
Comment (YuHsuan-Hwang) [reply src/Frame/Frame.cc]: fixed.

**Comment by YuHsuan-Hwang** [comment]
> Please also update protobuf changelog

This is updated.

**Review by confluence — APPROVED** [review]

**Review by confluence — COMMENTED** [review]
Please merge dev into this branch to resolve the merge conflict. Other than that I think this is fine.

**Review by pford — APPROVED** [review]

**Review by veggiesaurus — APPROVED** [review]

**Comment by YuHsuan-Hwang** [comment]
> Please merge dev into this branch to resolve the merge conflict. Other than that I think this is fine.

@confluence merged dev branch.

**Pull request merged**